pkg/boot/grub/grub.go

```diff
@@ -42,6 +42,8 @@ var probeGrubFiles = []string{
 	"boot/grub2/grub.cfg",
 }
 
+var GrubParser *parser
```

Comment:
two separate comments:

1) I'm always a little uncomfortable around global variables. Can we find a way to just pass the variables along to the BLS parser rather than use this global construct?

2) An exported global variable should not be exposed using an unexported type (`*parser`). Unexported types are not useful to users of the variable outside of the package. Is there a way to just export the variables, not the whole parser?

Reply:
I do see now where you're going with this... you're trying to have these parsers be completely independent. Let me see if I can get it straight:

#### what happens in the real world:

- GRUB is loaded from EFI or from MBR
- GRUB parses variables, etc
- GRUB then invokes its own BLS parser through `blscfg`, finds BLS files on potentially another partition (iirc from CentOS/RHEL 8 they're elsewhere, right?), parses those with `blscfg` in mind?

#### what you're proposing

- NERF comes up, does not know whether GRUB or BLS are present or prioritized in EFI or MBR
- NERF parses grub first (assuming there's only one grub config?), places grub variables in global variable
- NERF parses BLS next, independently from GRUB, and happens to interpret GRUB configs with GRUB variables in mind

this hinges on two properties that I'm not sure we want to promise: 

a) that GRUB is parsed first, and 
b) that, if multiple partitions each with their own GRUB configs are present, the GRUB config with the right variables is parsed last and thus places the right variables in the right global variable for BLS to pick up.

is that about right?

Reply:
Or like you said, the grub parser should invokes its own BLS parser through blscfg, finds BLS files on potentially another partition, parses those with blscfg in mind.

Reply:
Nice discussion. Would we be able to get some help/guidance from Grub community on this? I suppose the Grub community are mainly from OS vendors, such as redhat?

Reply:
Add a commit for invoking BLS parser through 'blscfg' within GRUB parser, which finds BLS files on potentially another partition. I feel this should be the proper way.

Reply:
On our CentOS 8 with the pkg/mount fix from this PR, there could be duplicate boot entries being displayed since all /boot/efi/EFI/centos/grub.cfg, /boot/grub2/grub.cfg and /boot/loader/entries/* can be parsed and some of them are the same OS image, but that shouldn't be an issue.

Reply:
@rminnich how to remove "awaiting author" label? Sorry, I looked around but did not find my way. The testing from WW side is good. I recommend us merging it in, and optimizing it down the road.
@johnnylinwiwynn  Please rebase, then we would move forward with approval and merge.

Reply:
I have removed label 'awaiting author' and rebased now, the build_and_test failure is not related to this PR so I think we can merge it. Thanks.